src/devices/sound/k053260.cpp

```diff
@@ -169,13 +172,33 @@ TIMER_CALLBACK_MEMBER(k053260_device::update_state_outputs)
 	switch (m_timer_state)
 	{
 		case 0: m_sh1_cb(ASSERT_LINE); break;
-		case 1: m_sh1_cb(CLEAR_LINE); break;
+		case 1: m_sh1_cb(CLEAR_LINE); tim2_count(); break;
 		case 2: m_sh2_cb(ASSERT_LINE); break;
 		case 3: m_sh2_cb(CLEAR_LINE); break;
 	}
 	m_timer_state = (m_timer_state+1) & 3;
 }
 
+// TIM2 is implemented on the original chip as two counters in series, where the
+// first one may be bypassed with a test bit, which is not used by glfgreat at
+// least. These counters together produce a count that is 112 times slower than
+// the SH1 pin, taking it down from 56kHz to 500Hz. The period of 500Hz is 2ms,
+// hence the pin name TIM2 and also the signal name 2MS seen in glfgreat schematics.
+void k053260_device::tim2_count()
+{
+	bool over = m_tim2 >= 111;
+	if (over)
+	{
+		m_tim2 = 0;
+		m_tim2_cb(ASSERT_LINE);
+	}
+	else
+	{
+		m_tim2++;
+		m_tim2_cb(CLEAR_LINE);
```

Comment:
You really shouldn’t be calling the devcb every time, only when it changes – that’s going to be a drain on performance.

Reply:
From what I've seen, yeah it's a performance drain, when it's in the MHz range (eg. a CPU with a devcb after every opcode).